src/main/java/Duke.java

```diff
@@ -1,10 +1,333 @@
+                break;
+            case "event":
+                numOfTasks = commandEvent(currentTasks, numOfTasks, splitUserInput[1]);
+                break;
+            case "help":
+                commandHelp();
+                break;
+            default:
+                try {
+                    throw new DukeException("UNINTELLIGIBLE_COMMAND");
+                } catch (DukeException exception) {
+                    printSeparator();
+                    System.out.println(exception);
+                    printSeparator();
+                }
+            }
+        }
+    }
+
+    /**
+     * Prints all tasks in task list
+     *
+     * @param currentTasks Tasks Array of all tasks
+     */
+    private static void commandList(Task[] currentTasks, int numOfTasks) {
+        printSeparator();
+        if (numOfTasks == 0) {
+            System.out.println("You have zero task at hand!");
+        }
+        for (int i = 0; i < numOfTasks; i++) {
+            System.out.println((i + 1) + "." + addBrackets(currentTasks[i].getType()) +
+                    addBrackets(currentTasks[i].getStatusIcon()) + " " + currentTasks[i].getDescription());
+        }
+        printSeparator();
+    }
+
+    /**
+     * Adds a Tasks.Deadline task to the currentTasks Tasks.Task array
+     *
+     * @param currentTasks Tasks Array of all tasks
+     * @param numOfTasks   Number of tasks
+     * @param userInput    User input
+     * @return numOfTasks Updated number of tasks
+     */
+    private static int commandDeadline(Task[] currentTasks, int numOfTasks, String userInput) {
+        try {
+            /*If user input contains only one word*/
+            if (userInput.length() < 1) {
+                throw new DukeException("MISSING_DESCRIPTION");
+            }
+            /*Missing qualifier*/
+            if (!userInput.contains("/by")) {
+                throw new DukeException("MISSING_QUALIFIER");
+            }
+            String[] splitInfoAndDeadline = userInput.trim().split(DEADLINE_QUALIFIER, 2);
+            /*If user input has no description*/
+            if (splitInfoAndDeadline[0].isEmpty()) {
+                throw new DukeException("MISSING_DESCRIPTION");
+            }
+            /*If user input has no deadline*/
+            if (splitInfoAndDeadline[1].isEmpty()) {
+                throw new DukeException("MISSING_INFO");
+            }
+            currentTasks[numOfTasks] = new Deadline(splitInfoAndDeadline[0], splitInfoAndDeadline[1]);
+            printSeparator();
+            System.out.println("Got it. I've added this task: \n" + "  " + addBrackets(currentTasks[numOfTasks].getType())
+                    + addBrackets(currentTasks[numOfTasks].getStatusIcon()) + " " + splitInfoAndDeadline[0] + " (by: " +
+                    splitInfoAndDeadline[1] + ")\n" + "Now you have " + (numOfTasks + 1) + (numOfTasks == 0 ? " task" :
+                    " tasks") + " in the list");
```

Comment:
Maybe you can try extracting this out into a method to improve readability?

Reply:
Thank you for this suggestion!